soql-types/src/main/scala/com/socrata/soql/types/obfuscation/CryptProvider.scala

```diff
@@ -5,8 +5,9 @@ import java.security.SecureRandom
 import org.bouncycastle.crypto.params.KeyParameter
 import org.bouncycastle.crypto.engines.BlowfishEngine
 
-class CryptProvider(key: Array[Byte]) {
-  private val keyParam = new KeyParameter(key)
+class CryptProvider(keyMaterial: Array[Byte]) {
+  def key = java.util.Arrays.copyOf(keyMaterial, keyMaterial.length)
```

Comment:
Why are you doing the this copy-ing?

Reply:
Because Java arrays are mutable.  It's just a defensive copy (and it's only 72 bytes, so not very expensive at all)